python/tests/kat/t_tcpmapping.py

```diff
@@ -264,3 +265,868 @@ def check(self):
 
             assert backend_fqdn == wanted_fqdn, f'{idx}: backend {backend_fqdn} != expected {wanted_fqdn}'
             assert tls_enabled == tls_wanted, f'{idx}: TLS status {tls_enabled} != wanted {tls_wanted}'
+
+
+class TCPMappingBasicTest(AmbassadorTest):
+    extra_ports = [ 6789 ]
+    target: ServiceType
+
+    def init(self) -> None:
+        self.target = HTTP()
+
+    def manifests(self) -> str:
+        return format('''
+---
+apiVersion: getambassador.io/v2
+kind: TCPMapping
+metadata:
+  name: {self.name.k8s}
+spec:
+  ambassador_id: [ {self.ambassador_id} ]
+  port: 6789
+  service: {self.target.path.fqdn}:80
+''') + super().manifests()
+
+    def queries(self):
+        yield Query(self.url("", port=6789))
+
+    def check(self):
+        assert self.results[0].json["backend"] == self.target.path.k8s
+        assert self.results[0].json["request"]["tls"]["enabled"] == False
+
+
+class TCPMappingCrossNamespaceTest(AmbassadorTest):
+    extra_ports = [ 6789 ]
+    target: ServiceType
+
+    def init(self) -> None:
+        self.target = HTTP(namespace="other-namespace")
+
+    def manifests(self) -> str:
+        return namespace_manifest("other-namespace") + format('''
+---
+apiVersion: getambassador.io/v2
+kind: TCPMapping
+metadata:
+  name: {self.name.k8s}
+spec:
+  ambassador_id: [ {self.ambassador_id} ]
+  port: 6789
+  service: {self.target.path.fqdn}:80
+''') + super().manifests()
+
+    def queries(self):
+        yield Query(self.url("", port=6789))
+
+    def check(self):
+        assert self.results[0].json["backend"] == self.target.path.k8s
+        assert self.results[0].json["request"]["tls"]["enabled"] == False
+
+
+class TCPMappingTLSOriginationBoolTest(AmbassadorTest):
+    extra_ports = [ 6789 ]
+    target: ServiceType
+
+    def init(self) -> None:
+        self.target = HTTP()
+
+    def manifests(self) -> str:
+        return format('''
+---
+apiVersion: getambassador.io/v2
+kind: TCPMapping
+metadata:
+  name: {self.name.k8s}
+spec:
+  ambassador_id: [ {self.ambassador_id} ]
+  port: 6789
+  service: {self.target.path.fqdn}:443
+  tls: true
```

Comment:
`tls: true` was how you said "originate TLS without a client cert (and with the default settings)" in v2; in v3alpha1 you do this by adding `https://` to the `service` (since allowing `tls` to be either a bool or a string is a violation of Kubernetes structural typing, which is something that v3alpha1 was going for).